azurelinuxagent/common/protocol/util.py

```diff
@@ -63,16 +64,14 @@ def get_protocol_util():
     return ProtocolUtil()
 
 
-class ProtocolUtil(object):
+class ProtocolUtil(SingletonPerThread):
     """
     ProtocolUtil handles initialization for protocol instance. 2 protocol types
     are invoked, wire protocol and metadata protocols.
     """
 
     def __init__(self):
-        self.lock = threading.Lock()
```

Comment:
Removed both locks since this is a SingletonPerThread class so we dont need to worry about concurrency issues (Additionally the SingletonPerThread has its own locks to ensure concurrency)